File: server/state.py
"""Shared runtime state for the server."""
from datetime import datetime, timezone
import os

DATA_DIR = os.getenv('DATA_DIR', 'data')

# Runtime collections
player_positions = {}
web_clients = set()
minecraft_connections = set()
block_events = []
active_players = set()
session_events = []
latest_assessment_results = {}

# Session tracking
session_start_time = None
session_id = None
session_file = None

# Event buffer
event_buffer = []
last_save_time = None


def ensure_data_directory():
    if not os.path.exists(DATA_DIR):
        os.makedirs(DATA_DIR)


def now_utc():
    return datetime.now(timezone.utc)

File: electron/server/state.ts
import type { WebSocket } from 'ws';

// Shared in-memory state between Minecraft and Web socket handlers
interface SessionEvent {
  timestamp: string;
  event_type: string;
  data?: any;
}

interface State {
  minecraftConnections: Set<WebSocket>;
  webClients: Set<WebSocket>;
  activePlayers: Set<string>;
  playerPositions: Record<string, any>;
  sessionId: string | null;
  sessionStartTime: Date | string | null;
  sessionFile: string | null;
  sessionEvents: SessionEvent[];
  eventBuffer: SessionEvent[];
  lastSaveTime: number | null;
  latest_assessment_results?: any;
}

const state: State = {
  minecraftConnections: new Set<WebSocket>(),
  webClients: new Set<WebSocket>(),
  activePlayers: new Set<string>(),
  playerPositions: {},
  sessionId: null,
  sessionStartTime: null,
  sessionFile: null,
  sessionEvents: [],
  eventBuffer: [],
  lastSaveTime: null,
};

export default state;
